app/src/main/java/com/saotome/imed_app/ui/fragments/DrugListFragment.kt

```diff
@@ -5,47 +5,43 @@ import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import com.saotome.imed_app.R
 import com.saotome.imed_app.databinding.FragmentDrugListBinding
 import com.saotome.imed_app.model.Drug
 import com.saotome.imed_app.ui.DrugListAdapter
 
-/**
- * A simple [Fragment] subclass as the default destination in the navigation.
- */
+
 class DrugListFragment : Fragment() {
 
     private var _binding: FragmentDrugListBinding? = null
-    private val drugList = arrayListOf<Drug>()
+    lateinit var drugListViewModel: DrugListViewModel
 
     private val adapter by lazy { DrugListAdapter() }
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-
-        loadDrugMockList()
-        //set loaded Drug List to adapter
-        adapter.submitList(drugList)
+        initData()
 
         _binding = FragmentDrugListBinding.inflate(inflater, container, false)
         _binding?.rvDrugs?.adapter = adapter
-
         return _binding?.root
     }
 
-    private fun loadDrugMockList() {
-        drugList.add(0, Drug(0, "Unak", "1 capsula", "20:00", "24:00"))
-        drugList.add(0, Drug(0, "Glifage", "1 capsula", "20:00", "24:00"))
-        drugList.add(0, Drug(0, "Indapen", "1 capsula", "10:00", "48:00"))
+    private fun initData() {
```

Comment:
Move the ViewModel's initialization to onCreateView. 

This function is called "initData" so it should only have logic related to data, not initializing the view model.